medic/medic-run.js

```diff
@@ -324,23 +349,8 @@ function main() {
         util.fatal("app " + appPath + " does not exist");
     }
 
-    util.medicLog("checking if " + couchdbURI + " is up");
-
-    // check if results server is up
-    request({
-        uri:     couchdbURI,
-        method:  "GET",
-        timeout: SERVER_RESPONSE_TIMEOUT
-    },
-    function (error, response, body) {
-
-        // bail if the results server is down
-        if (error || response.statusCode !== 200) {
-            util.fatal("it's not up, so test run can't be monitored");
-            process.exit(1);
-        } else {
-            util.medicLog("it's up");
-        }
+    onConnect(couchdbURI, 1, function(){
```

Comment:
If possible, maybe also change the signature of the retry function to hide the `1` from the caller, so that it only takes a URI and a callback.